command_line/sim_view.py

```diff
@@ -526,12 +575,13 @@ def _toggle_Fhkl(self, _press=None):
         self._display()
 
     def _toggle_spectrum_shape(self, _press=None):
-        options = ["Gaussian", "SASE", "monochromatic"]
-        current = options.index(self.spectrum_shape)
-        self.spectrum_shape = options[(current+1)%3]
-        self._update_spectrum(init=(self.spectrum_shape == "SASE"))
-        self._generate_image_data()
-        self._display()
+        if not self.diffuse_scattering:
```

Comment:
May be prudent to add a matching reset to monochromatic when diffuse is turned on.